lib/fastlane/plugin/sentry/actions/sentry_create_deploy.rb

```diff
@@ -0,0 +1,103 @@
+module Fastlane
+  module Actions
+    class SentryCreateDeployAction < Action
+      def self.run(params)
+        require 'shellwords'
+
+        Helper::SentryHelper.check_sentry_cli!
+        Helper::SentryConfig.parse_api_params(params)
+
+        version = params[:version]
+        version = "#{params[:app_identifier]}@#{params[:version]}" if params[:app_identifier]
+
+        has_time = !params[:time].nil?
+        has_started = !params[:started].nil?
+        has_finished = !params[:finished].nil?
+
+        if !has_time && (has_started && !has_finished || !has_started && has_finished)
+          UI.user_error!("Only one parmeter of 'started' and 'finished' found, please provide both.")
+        end
+
+        command = [
+          "sentry-cli",
+          "releases",
+          "deploys",
+          version,
+          "new"
+        ]
+        command.push('--env').push(params[:env]) unless params[:env].nil?
+        command.push('--name').push(params[:name]) unless params[:name].nil?
+        command.push('--url').push(params[:deploy_url]) unless params[:deploy_url].nil?
+        command.push('--started').push(params[:started]) unless params[:started].nil? || has_time
+        command.push('--finished').push(params[:finished]) unless params[:finished].nil? || has_time
+        command.push('--time').push(params[:time]) unless params[:time].nil?
+
+        Helper::SentryHelper.call_sentry_cli(command)
```

Comment:
How does this handle error? Does it throw, print any message from the command to the console and exists?

Reply:
I did have an error as provided the wrong format for the url to sentry-cli. It printed the error in the command line, we use the provided `fastlane` API for that `UI.user_error...` etc afaik.